src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs

```diff
@@ -207,6 +207,7 @@ private static bool GetIsInvariantGlobalization()
         public static bool IsIcuGlobalization => ICUVersion > new Version(0,0,0,0);
         public static bool IsNlsGlobalization => IsNotInvariantGlobalization && !IsIcuGlobalization;
 
+        [DynamicDependency("GetICUVersion", "Interop/Globalization", "System.Private.CoreLib")]
```

Comment:
I don't think this is going to work because the test code isn't linked when we trim System.Private.CoreLib.

Instead, I think we can move the entry in `ILLink.Descriptors.Shared.xml` to a ".LibraryBuild.xml" file that is only used during the build of the library. That way the `GetICUVersion` will be there when we build the library, but it can get trimmed at app trimming time.

Reply:
Here's other examples of "LibraryBuild.xml" files:

https://github.com/dotnet/runtime/blob/master/src/libraries/System.Diagnostics.StackTrace/src/ILLink/ILLinkTrim_LibraryBuild.xml
https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.Xml/src/ILLink/ILLinkTrim_LibraryBuild.xml

and the code to handle them for libraries (which will need to be updated for CoreLib):

https://github.com/dotnet/runtime/blob/515bfafb35f31ea689917c20b8e6a54c7e4a25d4/eng/illink.targets#L40-L41

Reply:
See also https://github.com/dotnet/runtime/issues/39102

Reply:
Added it.